pkg/dcgmexporter/pipeline_test.go

```diff
@@ -43,3 +46,68 @@ func TestRun(t *testing.T) {
 	// We'll be testing them more throughly in the e2e tests (e.g: by running prometheus).
 	t.Logf("Pipeline result is:\n%v", out)
 }
+
+func testNewDCGMCollector(counter *int, nvSwitch, link bool) DCGMCollectorConstructor {
+	return func(c []Counter, config *Config, entityType dcgm.Field_Entity_Group) (*DCGMCollector, func(), error) {
+		if entityType == dcgm.FE_SWITCH && !nvSwitch {
+			return nil, func() {}, fmt.Errorf("Switch collector should not be created")
+		}
+		if entityType == dcgm.FE_LINK && !link {
+			return nil, func() {}, fmt.Errorf("Link collector should not be created")
+		}
+
+		collector := &DCGMCollector{}
+		cleanups := []func(){
+			func() {
+				*counter++
+			},
+		}
+		collector.Cleanups = cleanups
+
+		return collector, func() { collector.Cleanup() }, nil
+	}
+}
+
+func TestCountPipelineCleanup(t *testing.T) {
+	emptyCSV := "./empty.csv"
```

Comment:
Please put this file into the temporary directory (for example https://pkg.go.dev/os#CreateTemp) to avoid leftovers after the testing.

Reply:
fixed